server/py/services/api/crud/runtimes/nuclio/function.py

```diff
@@ -331,12 +331,21 @@ def _compile_function_config(
             # if base_spec was not set (when not using code_to_function) and we have base64 code
             # we create the base spec with essential attributes
             config = nuclio.config.new_config()
-            mlrun.utils.update_in(config, "spec.handler", handler or "main:handler")
 
+        # if base_spec comes from client side, make sure that labels and annotations exists in dictionary
+        for key in [
+            "metadata.labels",
+            "metadata.annotations",
+        ]:
+            if not mlrun.utils.get_in(config, key):
+                mlrun.utils.update_in(config, key, {})
```

Comment:
what about handling it in `_set_function_metadata` and calling it above the if statement (line 321)?